### DISCUSSION TITLE
Is it safe to delete and automatically re-generate bucket metadata?

### DISCUSSION THREAD

**Discussion by omarkohl** [minio/minio#16096] [Q&A] [answered]
I have a setup with two sites and server-side site replication between them.

I configured and activated KES to later enable SSE-S3 encryption for my buckets. Then I made some mistakes and ended up overwriting my Filesystem KMS encryption key. At first I though this would be no problem because I had not activated the encryption for any bucket, but it turns out that MinIO immediately and proactively encrypts following `BucketMetadata` values:

```go
	BucketTargetsConfigJSON     []byte
	BucketTargetsConfigMetaJSON []byte
```

Therefore the bucket metadata is now corrupt and unreadable from MinIO's point of view.

On a test system I did the following.

1. Export the bucket metadata with `mc admin cluster bucket export`.
2. Corrupt the system in the same way.
3. Delete the bucket metadata (namely delete the directory `.minio.sys/buckets/BUCKET`).
4. Remove all the KES stuff from the config.
5. Restart MinIO.
6. Enable versioning (needed because of site-replication).
7. Export the bucket metadata again.

If I compare the bucket metadata from step 1 and step 7, they are identical except:

```diff
-    <Arn>arn:minio:replication::d9c03912-f098-4d4a-9b08-fcccc39adf11:normal</Arn>
+    <Arn>arn:minio:replication::33d75cbf-d4bb-4d1b-b04f-bc421766b03d:normal</Arn>
```

Other than that everything looks great.

I do however also see with xl-meta that individual objects also store the above ARN, but MinIO seems to overwrite/fix it at some point with the new value.

My question therefore is: Is this a good procedure? What are the risks? In particular, is it a problem that the ARN UUID changed?

**Comment by klauspost** [comment, marked as answer]
You should *never* modify backend data. Your answer is as short as that.

I will no go into speculation of what this can cause, since you shouldn't be doing it in the first place.

**Reply by omarkohl** [reply]
I absolutely agree with you under any other circumstances. But the fact is I have encrypted values in BucketMetadata and it's impossible for me to recover them. So I need to do something...

**Reply by omarkohl** [reply]
So given the following:

* This was entirely my mistake
* Neither I nor anyone else reading this should ever edit backend data
* Whatever I end up doing is my own responsibility and you are in no way responsible for giving me potentially bad advice

Would you be willing to answers some questions?

* Where is the ARN value correctly updated within the objects? What process does this? Can I trigger/force it somehow?
* Does the ARN have any other purposes? From reading the code I only deduce that it's an entirely random string that does not exist on the remote site but is just used as a site-local ID to identify remote buckets.
* Can you think of any other risks? You don't need to supply an exhaustive list and again, even if you forget something, it's my responsibility, not yours

**Reply by klauspost** [reply]
I don't have the answers you seek. Finding them would not be trivial, and could only do that if you have a [subscription](https://min.io/pricing).
